target/linux/ramips/files/drivers/dma/mediatek/hsdma-mt7621.c

```diff
@@ -465,7 +465,7 @@ static struct dma_async_tx_descriptor *mtk_hsdma_prep_dma_memcpy(
 	if (len <= 0)
 		return NULL;
 
-	desc = kzalloc(sizeof(*desc), GFP_ATOMIC);
+	desc = kzalloc(sizeof(*desc) + 1, GFP_ATOMIC);
```

Comment:
Why do you change this?

Reply:
The difference between [0] and [] is 1. I added it to keep the allocation the same.

Reply:
`[]` is an array with variable size. if it was allocated like this we probably assume that it is empty.